internal/characters/xianyun/plunge.go

```diff
@@ -0,0 +1,249 @@
+			Abil:       fmt.Sprintf("Driftcloud Wave %v", c.skillCounter),
+			AttackTag:  attacks.AttackTagPlunge,
+			ICDTag:     attacks.ICDTagNone,
+			ICDGroup:   attacks.ICDGroupDefault,
+			StrikeType: attacks.StrikeTypeDefault,
+			Element:    attributes.Anemo,
+			Durability: 25,
+			Mult:       leap[skillInd][c.TalentLvlSkill()],
+		}
+		snap := c.Snapshot(&ai)
+		c.c6mod(&snap)
+
+		c.Core.QueueAttackWithSnap(ai, snap, skillArea, 0, c.particleCB(), c.a1cb(), c.c4cb())
+
+		// reset window after leap
+		c.DeleteStatus(skillStateKey)
+		c.skillCounter = 0
+		c.skillEnemiesHit = nil
+		c.skillSrc = noSrcVal
+	}, skillHitmark)
+
+	return action.Info{
+		Frames:          frames.NewAbilFunc(driftcloudFrames[skillInd]),
+		State:           action.PlungeAttackState,
+		AnimationLength: driftcloudFrames[skillInd][action.InvalidAction],
+		CanQueueAfter:   driftcloudFrames[skillInd][action.ActionSwap],
+	}, nil
+}
+
+// Low Plunge attack damage queue generator
+// Use the "collision" optional argument if you want to do a falling hit on the way down
+// Default = 0
+func (c *char) LowPlungeAttack(p map[string]int) (action.Info, error) {
+	// last action must be skill (for leap)
+	// dont need to check airborne for this because she can plunge if she's on the ground anyways
+	if c.StatusIsActive(skillStateKey) {
+		return c.driftcloudWave(p)
+	}
+
+	switch c.Core.Player.Airborne() {
+	case player.AirborneVenti:
+		return action.Info{}, fmt.Errorf("%s low_plunge while airborne due to venti hold E is unimplemented due to lack of frame data. Please see https://docs.gcsim.app/mechanics/frames for how to contribute", c.Base.Key.String())
+	case player.AirborneXianyun:
+		return c.lowPlungeXY(p)
+	default:
+		return action.Info{}, fmt.Errorf("%s low_plunge can only be used while airborne", c.Base.Key.String())
+	}
+}
+
+func (c *char) lowPlungeXY(p map[string]int) (action.Info, error) {
+	if c.Core.Player.CurrentState() != action.JumpState {
+		return action.Info{}, errors.New("only plunge after using jump")
+	}
+
+	collision, ok := p["collision"]
+	if !ok {
+		collision = 0 // Whether or not Xianyun does a collision hit
+	}
+
+	if collision > 0 {
+		c.plungeCollision(collisionHitmark)
+	}
+
+	lowPlungeRadius := 3.0
+
+	ai := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Low Plunge",
+		AttackTag:  attacks.AttackTagPlunge,
+		ICDTag:     attacks.ICDTagNone,
+		ICDGroup:   attacks.ICDGroupDefault,
+		StrikeType: attacks.StrikeTypeDefault,
+		Element:    attributes.Anemo,
+		Durability: 25,
+		Mult:       lowPlunge[c.TalentLvlAttack()],
+	}
+	c.Core.QueueAttack(
+		ai,
+		combat.NewCircleHitOnTarget(c.Core.Combat.Player(), nil, lowPlungeRadius),
+		lowPlungeHitmark,
+		lowPlungeHitmark,
+	)
+
+	return action.Info{
+		Frames:          frames.NewAbilFunc(lowPlungeFramesXY),
+		AnimationLength: lowPlungeFramesXY[action.InvalidAction],
+		CanQueueAfter:   lowPlungeFramesXY[action.ActionDash],
+		State:           action.PlungeAttackState,
+	}, nil
```

Comment:
similar comment as the other PR. need to reset Airborne flag somewhere/somehow after each plunge